src/dataflow/src/source/s3/metrics.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use lazy_static::lazy_static;
+use prometheus::core::AtomicU64;
+use prometheus::{register_uint_counter_vec, DeleteOnDropCounter, UIntCounterVec};
+
+pub(super) struct BucketMetrics {
+    objects_downloaded: DeleteOnDropCounter<'static, AtomicU64>,
+    pub objects_duplicate: DeleteOnDropCounter<'static, AtomicU64>,
```

Comment:
Why is this one `pub`? Is this intended to be used outside of this file? I don't see it mentioned elsewhere in the review and `inc` is not incrementing this counter.

Reply:
it is used in `source/s3.rs` on line 206, I've added a comment explaining why it's different.